### ISSUE TITLE
tf.train.start_queue_runners cannot run under GPU？

### ISSUE THREAD

**Issue by juventi** [tensorflow/tensorflow#14021]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Python version**: 
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

python 3.5
cuda 8.0 / cudnn 5.1
tensorflow 1.2.1
gpu memory Usage > 95%
gpu Load < 30%
cpu load > 85%

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

When using tf.train.start_queue_runners in trainning,  the trainning process automatically 
goes into cpu mode, when use tf.device('/gpu:0') an error accur thats could not find the data in queue;
so, how should i use queue_runners+tfrecords in GPU?

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

Main Trainning Code:

with tf.Session() as sess:
    #with tf.device('/gpu:0')
    sess.run(init)
    saver = tf.train.Saver()

    coord = tf.train.Coordinator() 
    threads = tf.train.start_queue_runners(sess=sess, coord = coord)
    
    global_step = 1
   
    if isTrain: 
        tf.train.write_graph(sess.graph_def, './MaxoutModel/', "model.pb", as_text=True)
        ckpt = tf.train.get_checkpoint_state('./MaxoutModel/')
        
        if ckpt and ckpt.model_checkpoint_path:
            saver.restore(sess, ckpt.model_checkpoint_path)
            print('Load Model OK!')
        else:
            print('Load Model Error!!')
            pass
    
        for i in range(100):
            
            # learning rate decay
            max_learning_rate = 0.02
            min_learning_rate = 0.0001
            decay_speed = 16000
            learning_rate = min_learning_rate + (max_learning_rate - min_learning_rate) * math.exp(-i/decay_speed)

            tra_images, tra_labels = sess.run([img_batch, label_batch])
            sess.run(train_step, {X: batch_X, Y_: batch_Y, lr: learning_rate,  keep_prob: 0.75})

            if i%20 == 0:
                cost = sess.run(cross_entropy, 
                        {X: tra_images, Y_: tra_labels , keep_prob: 1.0})
            if i%20 == 0 and i !=0 :    
                saver.save(sess,save_path='./BNModel/model.ckpt',global_step=global_step+1)
        saver.save(sess,save_path='./BNModel/model.ckpt',global_step=global_step+1)

**Comment by michaelisard** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
